library/L1_Peripheral/lpc40xx/pulse_capture.hpp

```diff
@@ -0,0 +1,334 @@
+      inline static const Pin kCapture2Channel1Pin               = Pin(0, 5);
+      inline static const CaptureChannelPartial_t kTimerPartial2 = {
+        .timer_register = LPC_TIM2,
+        .power_id       = SystemController::Peripherals::kTimer2,
+        .irq            = IRQn::TIMER2_IRQn,
+        .user_callback  = &timer2_isr,
+        .capture_pin0   = kCapture2Channel0Pin,
+        .capture_pin1   = kCapture2Channel1Pin,
+        .channel_number = &timer2_channel_number
+      };
+
+      inline static CaptureIsr timer3_isr = nullptr;
+      inline static CaptureChannelNumber timer3_channel_number =
+          CaptureChannelNumber::kChannel1;
+      inline static const Pin kCapture3Channel0Pin               = Pin(0, 23);
+      inline static const Pin kCapture3Channel1Pin               = Pin(0, 24);
+      inline static const CaptureChannelPartial_t kTimerPartial3 = {
+        .timer_register = LPC_TIM3,
+        .power_id       = SystemController::Peripherals::kTimer3,
+        .irq            = IRQn::TIMER3_IRQn,
+        .user_callback  = &timer3_isr,
+        .capture_pin0   = kCapture3Channel0Pin,
+        .capture_pin1   = kCapture3Channel1Pin,
+        .channel_number = &timer3_channel_number
+      };
+
+     public:
+      /// Structure that defines the capture channels associated with timer 0
+      inline static const CaptureChannel_t kCaptureTimer0 = {
+        .channel = kTimerPartial0, .isr = TimerHandler<kTimerPartial0>
+      };
+
+      /// Structure that defines the capture channels associated with timer 1
+      inline static const CaptureChannel_t kCaptureTimer1 = {
+        .channel = kTimerPartial1, .isr = TimerHandler<kTimerPartial1>
+      };
+
+      /// Structure that defines the capture channels associated with timer 2
+      inline static const CaptureChannel_t kCaptureTimer2 = {
+        .channel = kTimerPartial2, .isr = TimerHandler<kTimerPartial2>
+      };
+
+      /// Structure that defines the capture channels associated with timer 3
+      inline static const CaptureChannel_t kCaptureTimer3 = {
+        .channel = kTimerPartial3, .isr = TimerHandler<kTimerPartial3>
+      };
+    };  // struct Channel
+
+    /// Handler used to collect capture event status and pass to a user callback
+    static void TimerHandler(const CaptureChannelPartial_t & channel)
+    {
+      constexpr bit::Mask kCaptureInterruptFlagBit =
+          bit::CreateMaskFromRange(4, 5);
+      constexpr uint8_t kResetCaptureInterrupts = 0b11;
+
+      if (*channel.user_callback != nullptr)
+      {
+        CaptureStatus_t status;
+        status.flags = channel.timer_register->IR;
+
+        if (*channel.channel_number == CaptureChannelNumber::kChannel0)
+        {
+          status.count = channel.timer_register->CR0;
+        }
+        else
+        {
+          status.count = channel.timer_register->CR1;
+        }
+
+        (*channel.user_callback)(status);
+        }
```

Comment:
```suggestion
        }
```
Got a bit confused here and thought you were not clearing the interrupt flags here. Can you move this curly brace into the correct place.